File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Queue<Integer> queue = new LinkedList<>();
        for (int i = 0; i < priorities.length; i++) {
            queue.add(priorities[i]);
        }
        
        while (true) {
            int value = queue.poll();
            
            // 우선순위가 높은게 따로 있으면 뒤에 추가
            if (!queue.isEmpty() && value < Collections.max(queue)) {
                queue.offer(value);
            } else {
                answer++;
                
                if (location == 0) {
                    break;
                }
            }
            
            // location 옮겨주기
            if (location > 0) {
                location--;
            } else {
                location = queue.size() - 1;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque

def solution(priorities, location):
    answer = 0
    
    p_queue = deque()
    l_queue = deque()
    
    for i in range(len(priorities)):
        p_queue.append(priorities[i])
        l_queue.append(i)
    
    while p_queue:
        p = p_queue.popleft()
        l = l_queue.popleft()
        
        if not p_queue or p >= max(p_queue):
            answer += 1
            if l == location:
                break
        else:
            p_queue.append(p)
            l_queue.append(l)
    
    return answer